sched/group/group_env.c

```diff
@@ -0,0 +1,115 @@
+/****************************************************************************
+ * sched/group/group_env.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#ifndef CONFIG_DISABLE_ENVIRON
+
+#include <sched.h>
+#include <assert.h>
+
+#include "environ/environ.h"
+#include "sched/sched.h"
+#include "group/group.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: group_setupenv
+ *
+ * Description:
+ *   Copy the internal environment structure of a parent task to its newly
+ *   created child task's internal environment.
+ *
+ * Input Parameters:
+ *   tcb  - TCB of the newly created task, which is considered as a child of
+ *   the current task.
+ *   envp - Environment variables
+ *
+ * Returned Value:
+ *   Zero on success
+ *
+ ****************************************************************************/
+
+int group_setupenv(FAR struct task_tcb_s *tcb, FAR char * const *envp)
```

Comment:
It can be done, however my thinking was to keep the group access via group_functions instead of taking shortcut to env_dup(). Like for example in nxtask_init() -> group_allocate(...), group_setuptaskfiles(...), and so forth.

I don't have a very strong opinion on this, if you wish I can remove group_setupenv()

Reply:
but env_dup touch/access task_group_s too. All functions declared in ched/environ/environ.h require task_group pointer as the first argument, that's why I suggest to move or merge the new similar env function to environ.h.

Reply:
Sure, it is fine to me either way

Reply:
Done